lib/purescript-cst/src/Language/PureScript/CST/Convert.hs

```diff
@@ -540,13 +541,16 @@ convertDeclaration fileName decl = case decl of
     where
       -- truncate to 25 chars to reduce verbosity
       -- of name and still keep it readable
-      -- unique identifier will be appended to this name
-      -- in desugaring proces
+      -- name will be used to create a GenIdent
+      -- in desugaring process
       genName :: Text.Text
-      genName = "$_" <> Text.take 25 (className <> typeArgs) <> "_"
+      genName = Text.take 25 (className <> typeArgs)
 
       className :: Text.Text
-      className = N.runProperName $ qualName cls
+      className = foldMap (uncurry Text.cons . first toLower)
+                . Text.uncons
```

Comment:
Oh, I'm completely on board with that.

Maybe we should start a wiki page with our current/evolving notions of good style? I frequently find myself waffling between a style that is more idiomatic/traditional Haskell versus one which has practical benefits like this, and not having a dominant style in the code itself doesn't help. (Another example: equational declaration style versus lambda case—I think the latter is usually superior where applicable, again because renaming the function touches fewer lines.)

Reply:
I think that would be good 👍

Reply:
Rather than a wiki page, should it be included in the main repo?

Another thing we could add to that is to use record types whenever something has 3+ args.

Reply:
I'm thinking there isn't much advantage to versioning the standard along with the code—particularly while we're hammering out what the standard is, that could generate a lot of noise in the commit history. Maybe if the wiki page stabilizes, it makes sense to add it then, and then gatekeep it with the same PR workflow we use for code. But for starting out, I think we want something that we can feel more fluid about rewriting.

Reply:
Using a wiki page initially and then merging it into the repo when it stabilizes sounds like a good tradeoff.